orderer/common/multichannel/registrar.go

```diff
@@ -237,6 +244,30 @@ func (r *Registrar) Initialize(consenters map[string]consensus.Consenter) {
 			logger.Panicf("Error initializing without a system channel: failed to find an etcdraft consenter")
 		}
 	}
+
+	if r.config.ChannelParticipation.Enabled {
+		// Initialize channel-participation API file repos
+		fileRepoDir := r.config.ChannelParticipation.FileRepoDir
+		logger.Infof("Channel Participation API enabled, registrar initializing with file repo %s", fileRepoDir)
+
+		if err := validateFileRepoDir(fileRepoDir); err != nil {
+			logger.Panicf("Error initializing file repo dir: %s", err)
+		}
```

Comment:
This is a good direction, but maybe it belongs within the file-repo constructor?